docs/architecture/overall.md

```diff
@@ -0,0 +1,73 @@
+# Overall Architecture
+
+Screwdriver is a collection of services that facilitate the workflow for
+Continuous Delivery pipelines.
+
+![Workflow](assets/workflow.png)
+
+## Workflow
+
+1. **Commit new code**
+
+    User starts a new build by one of the following operations:
+
+    - User pushes code to GitHub on the monitored branch
+    - User pushes code to GitHub on an open pull request to the monitored
+    branch
+    - User opens a new pull request on GitHub to the monitored branch
```

Comment:
I would switch the order between this one and the one before it; seems like a more logical progression to me.
Also, would the pull request be **to** the monitored branch or **against** it?